projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ProjectorService.kt

```diff
@@ -165,15 +161,18 @@ class ProjectorService : PersistentStateComponent<ProjectorConfig> {
   }
 
   private fun disable() {
-    if (confirmRestart("To disable Projector, restart is needed. Can I restart the IDE now?")) {
-      stateChanged()
-      autostart = false
-      restartIde()
-    }
+    AgentLauncher.stopServer(0)
+    enabled = EnabledState.STOPPED
+    stateChanged()
   }
 
   private fun enable() {
-    attachDynamicAgent()
+    if (enabled == EnabledState.HAS_VM_OPTIONS_AND_DISABLED) {
+      attachDynamicAgent()
+    }
+    else {
+      AgentLauncher.startServer()
```

Comment:
Do we expect `EnabledState.STOPPED` here? I suggest adding the corresponding check:
```suggestion
      check(enabled == EnabledState.STOPPED) { "Bad state: $enabled" }
      AgentLauncher.startServer()
```

Just to prevent possible bugs in the future if we add more states.